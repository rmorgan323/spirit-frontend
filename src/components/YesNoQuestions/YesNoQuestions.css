.YesNoQuestions {
  margin: 10px;
  width: 170px; }
  .YesNoQuestions .category-title {
    height: 72px;
    background-color: #eee;
    margin-bottom: 20px;
    padding: 5px 5px;
    text-align: center;
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 5px; }
