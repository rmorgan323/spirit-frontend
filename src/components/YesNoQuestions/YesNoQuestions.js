import React from 'react';
import YesNo from '../../containers/YesNo/YesNo';
import { PropTypes } from 'prop-types';
import './YesNoQuestions.css';

const YesNoQuestions = ({ title, questionsArray, databaseNamesArray }) => {
  const display = questionsArray.map((question, index) => {
    return (
      <YesNo
        question={question}
        databaseName={databaseNamesArray[index]}
        key={index}
      />
    );
  });

  return (
    <div className="YesNoQuestions">
      <h4 className="category-title">{title}</h4>
      {display}
    </div>
  );
};

export default YesNoQuestions;

YesNoQuestions.propTypes = {
  title: PropTypes.string,
  questionsArray: PropTypes.array,
  databaseNamesArray: PropTypes.array
};
