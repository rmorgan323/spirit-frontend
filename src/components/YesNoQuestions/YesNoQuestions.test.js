import React from 'react';
import YesNoQuestions from './YesNoQuestions';
import { shallow } from 'enzyme';
import {
  mockTitle,
  mockQuestionsArray,
  mockDatabaseNamesArray
} from '../../data/mockData/mockYesNoQuestionsData';

describe('YesNoQuestions tests', () => {
  it('should match snapshot', () => {
    const renderedApp = shallow(
      <YesNoQuestions
        title={mockTitle}
        questionsArray={mockQuestionsArray}
        databaseNamesArray={mockDatabaseNamesArray}
      />
    );

    expect(renderedApp).toMatchSnapshot();
  });
});
