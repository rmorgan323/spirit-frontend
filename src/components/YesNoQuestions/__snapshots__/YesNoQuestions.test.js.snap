// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YesNoQuestions tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <YesNoQuestions
    databaseNamesArray={
      Array [
        "pos_1_outer",
        "pos_1_inner",
      ]
    }
    questionsArray={
      Array [
        "Outer Core",
        "Inner Core",
      ]
    }
    title="Core Development"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4
          className="category-title"
        >
          Core Development
        </h4>,
        Array [
          <Connect(YesNo)
            databaseName="pos_1_outer"
            question="Outer Core"
          />,
          <Connect(YesNo)
            databaseName="pos_1_inner"
            question="Inner Core"
          />,
        ],
      ],
      "className": "YesNoQuestions",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Core Development",
          "className": "category-title",
        },
        "ref": null,
        "rendered": "Core Development",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "databaseName": "pos_1_outer",
          "question": "Outer Core",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "databaseName": "pos_1_inner",
          "question": "Inner Core",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4
            className="category-title"
          >
            Core Development
          </h4>,
          Array [
            <Connect(YesNo)
              databaseName="pos_1_outer"
              question="Outer Core"
            />,
            <Connect(YesNo)
              databaseName="pos_1_inner"
              question="Inner Core"
            />,
          ],
        ],
        "className": "YesNoQuestions",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Core Development",
            "className": "category-title",
          },
          "ref": null,
          "rendered": "Core Development",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "databaseName": "pos_1_outer",
            "question": "Outer Core",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "databaseName": "pos_1_inner",
            "question": "Inner Core",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
